util/log/file.go

```diff
@@ -221,3 +301,103 @@ func GetLogReader(filename string, allowAbsolute bool) (io.ReadCloser, error) {
 	}
 	return nil, err
 }
+
+// int64sortable is required so we can sort on an int64 slice.
+type int64sortable []int64
+
+func (a int64sortable) Len() int           { return len(a) }
+func (a int64sortable) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a int64sortable) Less(i, j int) bool { return a[i] < a[j] }
+
+// selectFiles selects all log files that have an timestamp before the endTime and
+// the correct severity. It then sorts them in decreasing order, with the most
+// recent as the first one.
+func selectFiles(logFiles []FileInfo, severity Severity, endTimestamp int64) []FileInfo {
+	logFileMap := make(map[int64]FileInfo)
+	var timestamps int64sortable
+	for _, logFile := range logFiles {
+		if logFile.Details.Severity != severity {
+			continue
+		}
+
+		timestamp := logFile.Details.Time.UnixNano()
+		if timestamp <= endTimestamp {
+			logFileMap[timestamp] = logFile
+			timestamps = append(timestamps, timestamp)
+		}
+	}
+
+	// Sort the files in reverse order so we will fetch the newest first.
+	sort.Sort(sort.Reverse(timestamps))
+
+	result := []FileInfo{}
+	for _, timestamp := range timestamps {
+		result = append(result, logFileMap[timestamp])
+	}
+	return result
+}
+
+// FetchEntriesFromFiles fetches all available logs on disk that are of the
+// level of severity (or worse) and are between the start and end times. It will
+// stop reading in new files if the EntiresCutoff is exceeded and no more files
+// will be retrieved. The logs entries returned will be in decreasing order,
+// with the closest log to the start time being the first entry.
+func FetchEntriesFromFiles(severity Severity, startTimestamp, endTimestamp int64) ([]proto.LogEntry, error) {
```

Comment:
`EntriesCutoff` should be a parameter to this function, so we can have a reasonable default but exceed it on demand.